### PR TITLE
Initial Roles API implementation

### LINKED ISSUE #36: Add Roles API
Part of #26.

This can be split out into multiple issues to reflect the endpoints in the API docs: http://docs.codecorpsapidevelop.apiary.io/#reference/categories
- [ ] List all
- [ ] Create

### PR DESCRIPTION
**Pull request by npendery**
Branched from skills so includes that code as well

Closes #36

### CODE REVIEW

**Comment by joshsmith** [comment]
Is this also in progress @npendery?

**Comment by joshsmith** [comment]
The error on the build is a little odd.

**Comment by npendery** [comment]
@JoshSmith yes, still in progress because of that ci error that I haven't been able to fully dig into

File: test/models/role_test.exs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+defmodule CodeCorps.RoleTest do
+  use CodeCorps.ModelCase
+
+  alias CodeCorps.Role
+
+  @valid_attrs %{ability: "Backend Development", kind: "technology", name: "Backend Developer"}
+  @invalid_attrs %{ability: "Juggling", kind: "circus", name: "Juggler"}
```
Comment (joshsmith) [inline test/models/role_test.exs:7]: I just laughed out loud at this. Thanks. 👍

**Pull request merged**